test/ssl-tests/04-client_auth.conf.in

```diff
@@ -39,153 +40,182 @@ sub generate_tests() {
-                            .($sctp ? "-sctp" : ""),
+		# Sanity-check simple handshake.
+		push @tests, {
+		    name => "server-auth-${protocol_name}"
+			    .($sctp ? "-sctp" : ""),
                     server => {
-                        "MinProtocol" => $protocol,
-                        "MaxProtocol" => $protocol
-                    },
-                    client => {
-                        "MinProtocol" => $protocol,
-                        "MaxProtocol" => $protocol
+			"MinProtocol" => $protocol,
+			"MaxProtocol" => $protocol
+		    },
+		    client => {
+			"MinProtocol" => $protocol,
+			"MaxProtocol" => $protocol
                     },
                     test   => {
-                        "ExpectedResult" => "Success",
-                        "Method" => $method,
+			"ExpectedResult" => "Success",
+			"Method" => $method,
                     },
-                };
-                $tests[-1]{"test"}{"UseSCTP"} = "Yes" if $sctp;
+		};
+		$tests[-1]{"test"}{"UseSCTP"} = "Yes" if $sctp;
 
-                # Handshake with client cert requested but not required or received.
-                push @tests, {
-                    name => "client-auth-${protocol_name}-request"
-                            .($sctp ? "-sctp" : ""),
-                    server => {
-                        "MinProtocol" => $protocol,
-                        "MaxProtocol" => $protocol,
-                        "VerifyMode" => "Request"
-                    },
-                    client => {
-                        "MinProtocol" => $protocol,
-                        "MaxProtocol" => $protocol
-                    },
-                    test   => {
-                        "ExpectedResult" => "Success",
-                        "Method" => $method,
-                    },
-                };
-                $tests[-1]{"test"}{"UseSCTP"} = "Yes" if $sctp;
+		for (my $post = 0; $post <= $postauthenabled; $post++) {
+		    # Handshake with client cert requested but not required or received.
+		    push @tests, {
+			name => "client-auth-${protocol_name}-request"
+                                .($sctp ? "-sctp" : "")
+				.($post ? "-post-handshake" : ""),
+                        server => {
+                            "MinProtocol" => $protocol,
+                            "MaxProtocol" => $protocol,
+                            "VerifyMode" => "Request"
+                        },
+                        client => {
+                            "MinProtocol" => $protocol,
+                            "MaxProtocol" => $protocol
+                        },
+                        test   => {
+                            "ExpectedResult" => "Success",
+                            "Method" => $method,
+                        },
+                    };
+                    $tests[-1]{"test"}{"UseSCTP"} = "Yes" if $sctp;
+		    if ($post) {
+			$tests[-1]{"server"}{"VerifyMode"} .= "PostHandshake";
```

Comment:
Is `.=` correct here or should it be just `=`. Same question for similar lines below

Reply:
Yes, when doing post-handshake auth (as indicated by `$post`) the VerifyMode needs to be updated. In this case the `Require` and `Request` method is retained, but appended with `PostHandshake` to end up with `RequirePostHandshake` and `RequestPostHandshake` which sets the appropriate flags via SSL_CONF.